src/NetworkEntities.js

```diff
@@ -156,15 +156,13 @@ class NetworkEntities {
     var entityList = [];
     for (var id in this.entities) {
       var entityCreator = NAF.utils.getCreator(this.entities[id]);
-      if (entityCreator === clientId) {
-        let persists;
-        const component = this.entities[id].getAttribute('networked');
-        if (component && component.persistent) {
-          persists = NAF.utils.isMine(this.entities[id]) || NAF.utils.takeOwnership(this.entities[id]);
-        }
-        if (!persists) {
-          var entity = this.removeEntity(id);
-          entityList.push(entity);
+      var entityOwner = NAF.utils.getNetworkOwner(this.entities[id]);
+      if (entityCreator === clientId || (!entityCreator && entityOwner === clientId)) {
+        if (this.entities[id].getAttribute("networked").persistent) {
```

Comment:
The previous version has a guard against `this.entities[id].getAttribute("networked")` being `falsey`
>  `if (component && component.persistent) {`

Not sure if the guard is necessary but pointing it out in case it was an unintentional change.

Reply:
Don't think its needed, but did not intend to make that change here.. Will leave in for now even though it seems overly defensive.